File: src/main/java/eu/qwsome/sql/condition/Not.java
package eu.qwsome.sql.condition;

public class Not implements Condition {

    private final Condition condition;

    Not(final Condition condition){
        this.condition = condition;
    }

    @Override
    public ValueConstructor getValues() {
        return this.condition.getValues();
    }

    @Override
    public void appendTo(final StringBuilder builder) {
        builder.append("NOT ");
        this.condition.appendTo(builder);
    }
}

File: src/main/kotlin/eu/qwsome/sql/condition/Not.kt
/*
 * @author Lukáš Kvídera
 */
package eu.qwsome.sql.condition

class Not(private val condition: Condition) : Condition {

    override fun getValues(): ValueConstructor = condition.getValues()

    override fun appendTo(builder: StringBuilder) {
        builder.append("NOT ")
        condition.appendTo(builder)
    }
}
